lib/src/SlashRemover.cc

```diff
@@ -0,0 +1,155 @@
+#include "drogon/plugins/SlashRemover.h"
+#include "drogon/HttpAppFramework.h"
+#include "drogon/utils/FunctionTraits.h"
+#include <functional>
+#include <string>
+
+using namespace drogon;
+using namespace plugin;
+using std::string;
+
+#define TRAILING_SLASH_REGEX ".+\\/$"
+#define DUPLICATE_SLASH_REGEX ".*\\/{2,}.*"
```

Comment:
For coding-wise, everything appears to be fine except for one thing. Instead of macros, I would rather use constexpr string_view (or const char*) instead of macros here.

Reply:
I agree with you, the reason for using `#define` is for the third case in the switch statement, so it concatenates both regexes at compile time. This is so we don't repeat both regexes in the third case.

Reply:
Oh, I missed that part. Then, macros would be a much neater way, which is great. Since this is not a header, we do not have to be that paranoid anyway.